### DISCUSSION TITLE
Torrent completions and images

### DISCUSSION THREAD

**Discussion by ghost** [torrust/torrust-index-gui#519] [Q&A] [answered]
Hey Guys,
Always glad to see the improvements to the project, and looking forward to that 3.0.0 due date. 
I played around with the live demo and had a couple questions.
1. Are there any plans to display the total downloads/completed downloads of a torrent? There's mention of it in the code, but I haven't yet seen it in the demo.
2. Looking at [this](https://index.torrust-demo.com/torrent/ec83837e5e828774d700dfa31636c40e89e079c3/the-floorwalker-charlie-chaplin) torrent, its description seems to contain an image which is only visible when one is signed-in, is this on purpose?

**Comment by josecelano** [comment, marked as answer]
Hi @ty5e3a45 thank you for you feedback!!

> Hey Guys, Always glad to see the improvements to the project, and looking forward to that 3.0.0 due date. 

We are getting closer: [roadmap](https://github.com/orgs/torrust/projects/10/views/1).

I played around with the live demo and had a couple questions.
> 
> 1. Are there any plans to display the total downloads/completed downloads of a torrent? There's mention of it in the code, but I haven't yet seen it in the demo.

We are not even importing those metrics from the tracker yet. We have them in the tracker but not in the Index (backend). However, I may be wrong because I've been more focused on the Tracker recently.

You can open a [proposal](https://github.com/torrust/torrust-index-gui/discussions/categories/proposals) for that. Anyway, it should be very easy to implement.

We have been focussed on the release v3.0.0, which only includes critical features. Our plan is to review the next [milestones](https://github.com/torrust/torrust-index-gui/milestones) after the release in a collaborative way. This is a simplified roadmap overview:

![image](https://github.com/torrust/torrust-index-gui/assets/58816/06cf1a81-6c4c-4b50-9eaa-709acac810f7)

I've tried to group and summarize features.

We are very interested in knowing what are the user's priorities.

> 2. Looking at [this](https://index.torrust-demo.com/torrent/ec83837e5e828774d700dfa31636c40e89e079c3/the-floorwalker-charlie-chaplin) torrent, its description seems to contain an image which is only visible when one is signed-in, is this on purpose?

Yes, I don't remember the exact reason now. I think it was to avoid DoS attacks because the Index acts as an image proxy to protect users' privacy. Since proxying images is a heavy task that consumes a lot of resources (bandwidth, CPU, and memory), we decided to show them only to authenticated users.

I will double-check it and add it to the docs if it's not included somewhere.

**Reply by ghost** [reply]
Appreciate the quick response!

**Reply by josecelano** [reply]
I've opened a feature request for point 1: https://github.com/torrust/torrust-index-gui/discussions/521

**Comment by josecelano** [comment]
Hi @ty5e3a45, I can improve my response for point 2 :-).

For now, the torrent list and torrent details pages are always public. It does not make sense to show all the info about the torrent except for the images in the description. However, as I told you, the image proxy consumes a lot of resources from the server, so we decided to assign a usage quota per user. That's the reason we need the user to be authenticated.

The image cache implementation: https://github.com/torrust/torrust-index/tree/develop/src/cache/image

In fact, there is a configuration section only for that image cache:

```toml
[image_cache]
max_request_timeout_ms = 1000
capacity = 128000000
entry_size_limit = 4000000
user_quota_period_seconds = 3600
user_quota_bytes = 64000000
```

Docs for image configuration: https://docs.rs/torrust-index/3.0.0-alpha.2/torrust_index/config/struct.ImageCache.html

**Reply by josecelano** [reply]
I've opened an [issue](https://github.com/torrust/torrust-index-gui/issues/522) to make it clear in our docs.
